test/e2e/config/e2e_conf.yaml

```diff
@@ -39,7 +39,7 @@ providers:
 - name: kubeadm
   type: BootstrapProvider
   versions:
-  - name: ${CAPI_FROM_RELEASE} # latest published release in the v1alpha4 series; this is used for v1alpha4 --> v1beta1 clusterctl upgrades test only.
+  - name: ${CAPI_FROM_RELEASE}
```

Comment:
I can't directly comment on the line, but what I am not understanding is why this is so much complicated rather then just using hardcoded values for CAPI_FROM and CAP_TO releases. That is not my main question though, but it is this line:

```
files:
    - sourcePath: "../data/shared/${CAPI_FROM_RELEASE:0:4}/metadata.yaml"
```

how it used to work AND will work as is? you have `v0.4`, `v1.2`, `v1.3`, `v1.4`, `v1.5`, `v1.6` folders under `data/shared` folder

Reply:
I agree this is hard to follow and confusing as well. We have another PR where we are hardcoding these values for simplifications, https://github.com/metal3-io/cluster-api-provider-metal3/pull/1465/, as for how it works, @mboukhalfa can clarify more, we are exporting `CAPI_TO_RELEASE` inherits its value from `CAPIRELEASE` which is set in dev-env. I think I have to also do a cleanup of the line you pointed out.

Reply:
that line is extract the the minor release and removing the patch chars to set the correct path based on the release exported in dev-env